src/Util/Images.php

```diff
@@ -418,19 +418,19 @@ public static function getScalingDimensions(int $width, int $height, int $max):
 
 			if ((($height * 9) / 16) > $width) {
 				$dest_width = $max;
-				$dest_height = intval(($height * $max) / $width);
+				$dest_height = intval(ceil(($height * $max) / $width));
```

Comment:
Why `ceil`? This would always increase the height by 1, wouldn't it? Wouldn't it be much easier and safer to use für example `max(intval($height * $max) / $width), 1);`?

Reply:
I guess it will mostly be one extra pixel compared to the existing approach, yeah.  But I don't see that as any less correct than what it's doing now, which is effectively `floor()`.  That is, `intval(479.9999)` will give `479`.

In general, as a user, to the extent I care about this at all, having one extra pixel gives me more image to look at.  `ceil()` very slightly biases the calculation to be very slightly "squarer", which is probably the right bias.

I do see adding an explicit rounding decision as a good thing for readability, rather than adding a special case for zero pixels.

And, if the input image really is 320x0, then the `max(..., 1)` approach will give a final height of 1.  Who knows what mess that might cause.

Reply:
I don't think that pictures can have `0` pixels, so this isn't an issue here. But you mentioned a good point. I guess that we should use the `round` function with this special case for the `0`. From a perfectionists standpoint I wouldn't appreciate that for example a `128x128` picture is resized to a `65x64` picture.

Reply:
I still think that adding another special case here makes the code more complicated and harder to maintain.  Especially in the absence of unit tests, I'd worry someone would undo the fix later.  There is already a pretty mind-bending special case of > 16:9 aspect ratio images.  Using `round` instead of `ceil` with `$mode = PHP_ROUND_HALF_UP` seems reasonable though - I'm probably too much of a C-programmer at heart 😜

Reply:
We do have got unit tests and it seems as if it should be no problem to add a test for `getScalingDimensions` in https://github.com/friendica/friendica/blob/develop/tests/src/Util/ImagesTest.php

Reply:
Yeah, you're right, I should do that before this gets merged. It especially could use some tests around the 16:9 special case too.

Reply:
Did you reply with your professional GitHub account? 👀

(Happened to me a few times)

Reply:
I actually have two professional accounts, but I lost access to the first one because I lost access to my 2FA seed and had no other way of proving account ownership 🤷‍♂️

Reply:
Argh yes - twice in one PR!  Thanks for the notice.  Not especially sensitive of course, but I do try to keep my employer's name out of it.  Anyway, the status is: I just got back from my travels, and on the weekend I plan to add unit tests.  This PR is on hold until that's done.

Reply:
@annando could you give me a hint how to run unit tests locally?  My notes say the last time I tried this was 3 years ago, but that's not working any more:
```
mat@aeon:~/Project/friendica$ USER=friendica_test PASS=friendica_test DB=friendica_test sudo phpunit --configuration tests/phpunit.xml
PHP Fatal error:  Uncaught Error: Class "Friendica\Test\FixtureTest" not found in /home/mat/Project/friendica/tests/functional/DependencyCheckTest.php:36
```
The file `tests/README.md` points to a non-existent script I'm supposed to run.

Reply:
It looks like it's now called `bin/dev/autotest.sh`.

Reply:
OK, thanks.  Tests added.